### PR TITLE
Update README after changing core to Freki

### PR DESCRIPTION
**Pull request by kalyan-kumar**
Refering to [Issue #40 ](https://github.com/mushorg/glutton/issues/40)

### CODE REVIEW

**Comment by glaslos** [comment]
Please remove all content in the README that is not relevant anymore.

File: README.md [hunk 1]
```diff
@@ -33,6 +33,23 @@ In case of error try
 service netfilter-persistent save  
 service netfilter-persistent reload
 ```
+Now, install dependencies using `glide`.
+```
+export PATH=$PATH:$GOPATH/bin
```
Comment (glaslos) [inline README.md]: This is not required if you have a proper Go environment.

File: README.md [hunk 2]
```diff
@@ -33,6 +33,23 @@ In case of error try
...
+curl https://glide.sh/get | sh
+glide install
```
Comment (glaslos) [inline README.md]: Why is this required?

File: README.md [hunk 3]
```diff
@@ -33,6 +33,23 @@ In case of error try
...
+glide update
+```
+Build glutton,
+```
+cd $GOPATH/src/github.com/mushorg/glutton
+go install app/server.go
```
Comment (glaslos) [inline README.md]: why `install`?

File: README.md (hunk 2 above)
Comment (kalyan-kumar) [inline README.md]: Installing glide so that dependencies could be installed from this. Just like we discussed on the issue.

File: README.md (hunk 3 above)
Comment (kalyan-kumar) [inline README.md]: Isn't that the way to compile the code and create the executable?

File: README.md (hunk 2 above)
Comment (glaslos) [inline README.md]: correct.

File: README.md (hunk 3 above)
Comment (glaslos) [inline README.md]: What is `go build` for? What does `go run` do?

Comment (kalyan-kumar) [inline README.md]: `go build` also does the same as `go install`, but from what I've read, `go build` doesn't cache anything. `go install` caches non-main packages so that the next compile happens faster. `go run` is for running the executable of the main package.

**Comment by kalyan-kumar** [comment]
@glaslos Please review now.

File: README.md [hunk 4]
```diff
@@ -2,17 +2,12 @@
 
 The Glutton server listens on both TCP and UDP port 5000 for new connections.
```
Comment (glaslos) [inline README.md]: This is not required anymore

File: README.md [hunk 5]
```diff
@@ -2,17 +2,12 @@
...
 ```
-apt-get install iptables-persistent conntrack golang libpcap-dev
-```
-Download and set up glutton, add GOPATH to /etc/environment. Example
-```
-mkdir /opt/go
-echo export GOPATH=/opt/go >> /etc/environment
-source /etc/environment
-go get github.com/mushorg/glutton
+apt-get install iptables-persistent conntrack libpcap-dev libnetfilter-queue-dev golang
```
Comment (glaslos) [inline README.md]: This is not correct anymore. Here are the packages required: go 1.7+, libnetfilter-queue-dev, libpcap-dev, iptables-dev

File: README.md [hunk 6]
```diff
@@ -2,17 +2,12 @@
...
 ```
+During installation of `iptables-persistent`, select YES for saving your current firewall rules for both ipv4 and ipv6. Make sure `go` environment is setup properly.
```
Comment (glaslos) [inline README.md]: I don't think this is required anymore.

File: README.md [hunk 7]
```diff
@@ -33,14 +28,31 @@ In case of error try
...
+```
+curl https://glide.sh/get | sh
+glide install
+glide update
+```
+Download and setup glutton,
+```
+go get github.com/mushorg/glutton
+cd $GOPATH/src/github.com/mushorg/glutton
+go install app/server.go
```
Comment (glaslos) [inline README.md]: This is not required.

File: README.md [hunk 8]
```diff
@@ -33,14 +28,31 @@ In case of error try
...
+mkdir /etc/glutton
+cp $GOPATH/src/github.com/mushorg/glutton/rules/rules.yaml /etc/glutton
```
Comment (glaslos) [inline README.md]: Why copy the rules.yaml file?

File: README.md [hunk 9]
```diff
@@ -33,14 +28,31 @@ In case of error try
...
+```
+cd $GOPATH/src/github.com/mushorg/glutton
+go run app/server.go
+```
 To test glutton
 ```
-mkdir -p /etc/glutton  
 mkdir -p /var/log/glutton
 go get github.com/google/gopacket
 go get pkg.re/essentialkaos/z7.v2
```
Comment (glaslos) [inline README.md]: No more `go get` with glide.

File: README.md (hunk 8 above)
Comment (kalyan-kumar) [inline README.md]: When I was trying to install, I got an error that this file wasn't found in /etc/glutton

File: README.md (hunk 7 above)
Comment (kalyan-kumar) [inline README.md]: This is the download and install step. How come this isn't needed?

File: README.md (hunk 5 above)
Comment (kalyan-kumar) [inline README.md]: Will change this accordingly.

File: README.md (hunk 9 above)
Comment (kalyan-kumar) [inline README.md]: Ohh yes. This will not be required anymore.

File: README.md (hunk 7 above)
Comment (glaslos) [inline README.md]: Why use `go run` if you did a `go install`?

File: README.md (hunk 8 above)
Comment (glaslos) [inline README.md]: `/etc/glutton/` is the default path.

**Comment by glaslos** [comment]
Can you please rebase your branch?

File: README.md (hunk 7 above)
Comment (kalyan-kumar) [inline README.md]: All right. Instead of `go run`, I'll simply run the executable.

**Comment by kalyan-kumar** [comment]
@glaslos Please review.

**Comment by glaslos** [comment]
All `iptables` related info is not required anymore.

**Comment by glaslos** [comment]
`glide install` and  `glide update` only works if you have the Glutton source code. Please fix the order.

File: README.md [hunk 10]
```diff
@@ -33,20 +24,30 @@ In case of error try
...
-cp $GOPATH/src/github.com/mushorg/glutton/scripts/glutton.conf /etc/init
+go get github.com/mushorg/glutton
+cd $GOPATH/src/github.com/mushorg/glutton
+go install app/server.go
+mkdir /etc/glutton
+cp $GOPATH/src/github.com/mushorg/glutton/rules/rules.yaml /etc/glutton
+```
+To run glutton,
+```
+cd $GOPATH/bin
```
Comment (glaslos) [inline README.md]: `$GOPATH/bin` is in your `$PATH` if you have set up Go correctly.

File: README.md [hunk 11]
```diff
@@ -33,20 +24,30 @@ In case of error try
...
+./app
```
Comment (glaslos) [inline README.md]: I don't like the fact that the binary is called `app`. please fix.

File: README.md [hunk 12]
```diff
@@ -33,20 +24,30 @@ In case of error try
...
+```
+To test glutton
+```
+mkdir -p /var/log/glutton
```
Comment (glaslos) [inline README.md]: Why are you creating this path if you are not using it?

File: README.md (hunk 11 above)
Comment (kalyan-kumar) [inline README.md]: Shall I name it `glutton`?

File: README.md (hunk 12 above)
Comment (kalyan-kumar) [inline README.md]: I am yet unaware of exact working of `glutton`. I just assumed that this is where the logs of `glutton` go into. Isn't this necessary then?

File: README.md (hunk 11 above)
Comment (glaslos) [inline README.md]: works for me

File: README.md (hunk 12 above)
Comment (glaslos) [inline README.md]: try it :) run glutton

File: README.md (hunk 11 above)
Comment (kalyan-kumar) [inline README.md]: I am naming it `gluttonserver` and moving all the glutton app code into a new sub directory `glutton`.

Comment (glaslos) [inline README.md]: Please don't change the directory name. There are better solutions.

Comment (kalyan-kumar) [inline README.md]: Wouldn't it be better if all the library code is put in a separate directory inside the project?

Comment (glaslos) [inline README.md]: Glutton is a package. server.go uses that package. I don't see an advantage in putting all the glutton code in a separate directory.

Comment (kalyan-kumar) [inline README.md]: All right. I understood. Let the library code be in the main package. But I think the folder `app` has to be renamed anyway. It is too generic. `gluttonserver` would be a better name, wouldn't it?

Comment (glaslos) [inline README.md]: I don't think it makes a difference. I wouldn't spend too much time thinking about it. It's also not related to the issue you want to solve with this pull request.

**Comment by kalyan-kumar** [comment]
@glaslos Please review :)

**Comment by glaslos** [comment]
I don't see any changes. Please don't change the branch. Update the current branch so I see what you do between the commits.

**Comment by kalyan-kumar** [comment]
I'm sorry, there was a problem when I pushed the commit. It is updated now. :)

**Comment by glaslos** [comment]
Okay, works for me now. Here are some recommendation:
- Read up on git workflow, don't undo git commits, add new commits to your branch/PR.
- When you get feedback, try to incorporate all the changes before asking for a new review.
- Especially for documentation, make sure you do all the steps on a fresh system to confirm they are correct.

Thanks for the contribution.
